libp2p/services/hpservice.nim

```diff
@@ -89,13 +89,18 @@ proc newConnectedPeerHandler(
   except CatchableError as err:
     debug "Hole punching failed during dcutr", err = err.msg
 
-method setup*(self: HPService, switch: Switch): Future[bool] {.async.} =
+method setup*(
+    self: HPService, switch: Switch
+): Future[bool] {.async: (raises: [CancelledError]).} =
   var hasBeenSetup = await procCall Service(self).setup(switch)
   hasBeenSetup = hasBeenSetup and await self.autonatService.setup(switch)
 
   if hasBeenSetup:
-    let dcutrProto = Dcutr.new(switch)
-    switch.mount(dcutrProto)
+    try:
+      let dcutrProto = Dcutr.new(switch)
+      switch.mount(dcutrProto)
+    except LPError as err:
+      trace "Failed to mount Dcutr", err = err.msg
```

Comment:
Should probably be an error. Trace's are not logged unless you explicitly set that that log level.
```suggestion
      error "Failed to mount Dcutr", err = err.msg
```

Also, does it make sense for the setup to continue Dcutr wasn't mounted? I guess it should be okay since this is used for direct connections when behind a nat :thinking:

Reply:
well, exceptions raised in `switch.mount` doesn't really help here, 
because `Dcutr.new(switch)` was created here, so we know that protocol is "mountable". 

what we can do is add `raiseAssert` here?